src/pages/Home/Home.tsx

```diff
@@ -0,0 +1,157 @@
+import { useState, useEffect, useRef } from "react";
+import ReactDOM from "react-dom";
+import { useTransition } from "react-spring";
+import LogoBar from "components/LogoBar";
+import NavBar from "./NavBar/NavBar";
+import Content from "./Content/Content";
+import Overlay from "components/Overlay";
+import MenuModal from "./Menu/MenuModal";
+import { HomeHeaderStyle } from "./HomeHeader.style";
+
+export type CategoryPosition = {
+  name: string;
+  offsetY: number;
+};
+
+export const HEADER_HEIGHT = 119;
+export const BORDER_TOP_WIDTH = 8;
+
+export default function Home() {
+  // TODO: Future implementation
+  // const [searchShow, setSearchShow] = useState(false);
+  // const [basketShow, setBasketShow] = useState(false);
+  // const [orderHistoryShow, setOrderHistoryShow] = useState(false);
+
+  const [menuShow, setMenuShow] = useState(false);
+  const [activeCategory, setActiveCategory] = useState("");
+  const [categories, setCategories] = useState<CategoryPosition[]>([]);
+  const [scrollPosition, setScrollPosition] = useState(0);
+  const [loaded, setLoaded] = useState(false);
+  const contentDivRef = useRef<HTMLDivElement>(null);
+
+  // onMount, add scrollListeners, Initialise categories, y-position, and default activeCategory.
+  useEffect(() => {
+    function scrollHandler() {
+      setScrollPosition(window.scrollY);
+    }
+    function loadHandler() {
+      setLoaded(true);
+    }
+    window.addEventListener("scroll", scrollHandler);
+    window.addEventListener("load", loadHandler);
+
+    let categoryAndPosition: CategoryPosition[] = [];
+
+    const SectionElements = contentDivRef.current?.getElementsByClassName(
+      "FoodCategorySectionHeader"
+    );
+    const length = SectionElements?.length ? SectionElements.length : 0;
+
+    //BUGFIX
+    if (!loaded) {
+      console.log("not loaded!");
+    } else {
+      console.log("loaded");
+    }
+
+    for (let i = 0; i < length; i++) {
+      let offsetY =
+        (SectionElements?.item(i)?.getBoundingClientRect().top as number) +
+        window.scrollY;
+      if (i !== 0) offsetY += BORDER_TOP_WIDTH;
+      const categoryName = SectionElements?.item(i)?.firstChild
+        ?.textContent as string;
+      categoryAndPosition.push({ name: categoryName, offsetY: offsetY });
+    }
+    console.log(categoryAndPosition);
+    setCategories(categoryAndPosition);
+    setActiveCategory(categoryAndPosition[0].name);
```

Comment:
This whole thing should be in loadHandler. You don't need to have a useState for whether the page has loaded or not as you're not using it. Just run the code when the page loads